test/trento_web/controllers/v1/host_controller_test.exs

```diff
@@ -48,4 +48,54 @@ defmodule TrentoWeb.V1.HostControllerTest do
              } == resp
     end
   end
+
+  describe "delete" do
+    test "should send 204 response when successful host deletion", %{conn: conn} do
+      %{id: host_id} = insert(:host)
+
+      expect(
+        Trento.Commanded.Mock,
+        :dispatch,
+        fn %Trento.Domain.Commands.RequestHostDeregistration{host_id: ^host_id} ->
+          :ok
+        end
+      )
+
+      conn
+      |> delete("/api/v1/hosts/#{host_id}")
+      |> response(204)
+    end
+
+    test "should send 422 response if the host is still alive", %{conn: conn} do
+      %{id: host_id} = insert(:host)
+
+      expect(
+        Trento.Commanded.Mock,
+        :dispatch,
+        fn %Trento.Domain.Commands.RequestHostDeregistration{host_id: ^host_id} ->
+          {:error, :host_alive}
+        end
+      )
+
+      conn
+      |> delete("/api/v1/hosts/#{host_id}")
+      |> response(422)
+    end
+
+    test "should return 404 if the host was not found", %{conn: conn} do
+      %{id: host_id} = insert(:host)
+
+      expect(
+        Trento.Commanded.Mock,
+        :dispatch,
+        fn %Trento.Domain.Commands.RequestHostDeregistration{host_id: ^host_id} ->
+          {:error, :host_not_registered}
+        end
+      )
+
+      conn
+      |> delete("/api/v1/hosts/#{host_id}")
+      |> response(:not_found)
```

Comment:
Not found is a json schema object, described by our openapi spec for that controller, please use the appropriate test helper from the openapi library